__tests__/app/(dashboard)/dashboard/chat/[chatId]/helpers.test.ts

```diff
@@ -0,0 +1,86 @@
+import {Helpers} from "@/app/(dashboard)/dashboard/chat/[chatId]/helpers";
+import {notFound} from "next/navigation";
+import fetchRedis from "@/helpers/redis";
+import {messageArraySchema} from "@/lib/validations/messages"
+
+jest.mock("@/helpers/redis", () => ({
+    __esModule: true,
+    default: jest.fn(),
+}));
+
+jest.mock("next/navigation", () => ({
+    notFound: jest.fn(),
+}));
+
+describe('getChatMessage Tests', () => {
+    let helpers: Helpers;
+    beforeEach(() => {
+        helpers = new Helpers();
+        jest.spyOn(helpers, 'fetchChatById').mockResolvedValue(["{\"id\": \"1701\", \"senderId\": \"bob\", \"text\":\"Hello world. My name's Gypsy, what's yours?\", \"timestamp\":1729437427}"]);
+
+    })
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    test('if fetchChatById throws, notFound is called', async () => {
+        jest.spyOn(helpers, 'fetchChatById').mockRejectedValue('Bad')
+
+        await helpers.getChatMessage("chat:userOne-userTwo:message");
+
+        expect(notFound).toHaveBeenCalled();
+    });
+
+    test("if fetchRedis doesn't throw, notFound isn't called", async () => {
+        await helpers.getChatMessage("chat:userOne-userTwo:message");
+
+        expect(notFound).not.toHaveBeenCalled();
+    });
+
+    test('If getChatMessage is called, then messageArraySchema.parse is called with the output of fetchChat',async ()=>{
+        jest.spyOn(helpers, "fetchChatById").mockResolvedValue(["{\"id\": \"1701\", \"senderId\": \"bob\", \"text\":\"Hello world. My name's Gypsy, what's yours?\", \"timestamp\":1729437427}"]);
+        const validatorSpy = jest.spyOn(messageArraySchema, 'parse');
+
+        await helpers.getChatMessage("chat:userOne-userTwo:message");
+
+        expect(validatorSpy).toHaveBeenCalledWith([{id: '1701', senderId: 'bob', text:"Hello world. My name's Gypsy, what's yours?", timestamp:1729437427}]);
+    })
+
+    test('if fetchByChatId returns an array with the most recent timestamp first, then return the array reversed', async()=>{
+        jest.spyOn(helpers, "fetchChatById").mockResolvedValue(["{\"id\": \"1701\", \"senderId\": \"bob\", \"recieverId\": \"1071\", \"text\":\"Hi Alice, what's up\", \"timestamp\":1729533949}",
+            "{\"id\": \"1071\", \"senderId\": \"alice\", \"recieverId\":\"1701\", \"text\":\"Hi bob\", \"timestamp\":1729437427}"]);
+        // @ts-expect-error rough mock for green parse
```

Comment:
- separate data from implementation
- it may be worth it to find a way to remove the ts-ignore, if it's not a huge rabbit hole